plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * REST API integration for the plugin.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Namespace for performance-lab REST API.
+ *
+ * @var string
+ */
+const PERFLAB_REST_API_NAMESPACE = 'performance-lab/v1';
+
+/**
+ * Route for activating plugin.
+ *
+ * @var string
+ */
+const PERFLAB_ACTIVATE_PLUGIN_ROUTE = '/activate-plugin';
```

Comment:
Since this endpoint is not a noun, it may make more sense as `/plugin:activate`. See for example in Optimization Detective:

https://github.com/WordPress/performance/blob/2bec0e8edbfbc3d8b2fb1216b40ec11988e4a61c/plugins/optimization-detective/storage/rest-api.php#L20-L30

```suggestion
/**
 * Route for activating plugin.
 *
 * Note the `:activate` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way 
 * that does not strictly follow the standard usage.
 * 
 * @link https://google.aip.dev/136 
 * @var string
 */
const PERFLAB_ACTIVATE_PLUGIN_ROUTE = '/plugin:activate;
```

Reply:
Indeed this is not a great resource URI for a REST URL, as it's not truly RESTful. REST endpoints should be centered around a resource, in this case a plugin.

But even better would be something like: `/plugins/<slug>:activate`. This would also be more intuitive in that the required parameter `slug` would be directly in the route path.
```suggestion
/**
 * Route for activating plugin.
 *
 * @var string
 */
const PERFLAB_ACTIVATE_PLUGIN_ROUTE = '/plugins/(?P<slug>[a-z0-9_-]+):activate';
```

Reply:
@felixarntz I was just curious what should be the criteria for deciding whether to add the slug to the URL itself or include it in the body.

Reply:
@b1ink0 It's one of the best practices for designing RESTful APIs, they should be centered around resources. See for example https://google.aip.dev/121 and https://google.aip.dev/122. (While these links are for Google's best practices, the best practices are not really specific to Google.)

By centering the API endpoints and their names around resources and their hierarchies, you future-proof the API to remain consistent and intuitive even when you add more and more features to it.

Here, the resource is a plugin. That's why using something like `/plugins/<slug>` as foundation for the routes is a good starting point. Such an endpoint could be used to get data for a specific plugin, or update data for a plugin for example.

Activating a plugin is not one of the CRUD operations though, so that's where a custom method is needed. See https://google.aip.dev/136 in that regard: That's how `/plugins/<slug>:activate` makes sense.

If we later needed a method to deactivate a plugin, that would be straightforward via `/plugins/<slug>:deactivate`. If we used `/activate-plugin`, this would need to be `/deactivate-plugin`. That's still somewhat consistent but not centered around the resource and less intuitive. For example, nothing in the name tells you that you have to provide a slug. Additionally, even if you guessed that some identifier for the plugin would be needed, you wouldn't know whether it needs to be a parameter called `slug` or `id` or `identifier` for example. The resource identifier being part of the endpoint path makes that easier to work with.

Reply:
@westonruter Following up on the other thread (since it probably makes sense to center that conversation about the endpoint naming): Per https://google.aip.dev/122, I think what would make most sense for the settings URL would be to simply be a value on the object returned by `GET /plugins/<slug>`.

A settings URL is not part of a collection, because a plugin does not have multiple settings URL. So `GET /plugins/<slug>/settings-url` would not be right. Because it's not an action (as you previously stated), `GET /plugins/<slug>:settings-url` wouldn't be right either.

The settings URL is one scalar piece of data for a plugin, so it makes most sense to be exposed on a plugin object. Obviously for this PR we don't care about creating a fully fledged `GET /plugins/<slug>` endpoint, but that's fine. We can simply for now return an object that e.g. only has `slug` and `settingsUrl` properties. The schema for such an object could be extended in the future in case we ever found a need for it, without backward compatibility breaks.

So I think going with `POST /plugins/<slug>:activate` and `GET /plugins/<slug>` (the latter of which return an object that includes the settings URL) is best in line with the API design guidelines.

Reply:
Okey thank you!

Reply:
@felixarntz there is already a [`plugins` endpoint](https://developer.wordpress.org/rest-api/reference/plugins/) in core, though. Perhaps we should add a new `perflab_settings_url` field to the plugin entity returned there?

Reply:
@felixarntz Additionally, there is also already a way to activate a plugin using the REST API by setting the `status` field  on the entity to `active`, without resorting to a special endpoint naming scheme: https://developer.wordpress.org/rest-api/reference/plugins/#update-a-plugin

However, reusing this endpoint is more complicated since we also automatically install the plugin prior to activation, as well as installing and activating any dependencies.

So maybe we should keep our own endpoint for installation and activation but instead of calling it `plugin:activate` we call it `feature:activate` to differentiate it from general plugin activation, as these plugins represent performance features. 

In the end, this API is almost guaranteed to be used 100% internally and so we don't really need to stress about it being getting it right forever.

Reply:
I don't think the existing plugin endpoints in Core are a problem. We use a different namespace specific to Performance Lab, so I wouldn't worry about that. It's IMO not unreasonable to provide custom endpoints for similar use-cases under a different namespace when the Core endpoints do not satisfy the requirements we have.

Regarding updating a `status` field to `active`, that's how they chose to build it, but doesn't seem that great to me per https://google.aip.dev/216.

That said, I'd be okay using the terminology "features" instead of "plugins" for our endpoints, since that's also what we use in the UI, and I agree it can help differentiate.

Reply:
So let's go with `POST /features/<slug>:activate` and `GET /features/<slug>` then?

Reply:
Where `GET /features/<slug>` returns an object with one key, `settingsUrl` which may be either a `string` or `null`, right? Sounds good.

Reply:
Exactly.

If we like, we could also include e.g. `slug` in the response object as that's simple enough and clarifies the intent of the endpoint even more (that it's supposed to return plugin information, so not necessarily _just_ the settings URL).